docs/src/tutorial/genomic-surveillance.rst

```diff
@@ -0,0 +1,158 @@
+     idaho:
+       title: "Idaho-specific genomic surveillance build"
+       subsampling_scheme: idaho_scheme
+       auspice_config: ncov-tutorial/auspice-config-custom-data.json
+
+   subsampling:
+     idaho_scheme:
+       custom_sample:
+         query: --query "(custom_data == 'yes')"
+         max_sequences: 5000
+       usa_context:
+         query: --query "(background_data != 'yes') & (country == 'USA')"
+         max_sequences: 1000
+         group_by: division year month
+         priorities:
+           type: proximity
+           focus: custom_sample
+       global_context:
+         query: --query "(reference_data == 'yes')"
+         max_sequences: 1000
+         priorities:
+           type: proximity
+           focus: custom_sample
+
+This is similar to the previous file, with some differences:
+
+1. The file paths in the second input are changed to ``idaho.metadata.tsv`` and ``idaho.sequences.fasta``.
+2. There is an additional input ``background_data`` for a regional North America dataset built by the Nextstrain team, for additional context.
+3. The output dataset is renamed ``idaho``, representative of the new custom data in the second input.
+4. There is a new ``subsampling`` section that provides a subsampling scheme ``idaho_scheme`` consisting of three subsamples:
+
+   1. ``custom_sample``
+
+      - This selects sequences from the ``custom_data`` input, up to a maximum of 5000 sequences.
+
+   2. ``usa_context``
+
+      - This selects sequences from the ``background_data`` input, up to a maximum of 1000 sequences.
+      - Sequences are subsampled evenly across all combinations of ``division``, ``year``, ``month``, with sequences genetically similar to ``custom_sample`` prioritized over other sequences.
+
+   3. ``global_context``
+
+      - This selects sequences from the ``reference_data`` input.
+
+This configuration file works without modifications, but the following steps guide you to customize names and titles, if appropriate:
+
+1. Change the ``custom_data`` input filenames from ``idaho.metadata.tsv`` and ``idaho.sequences.fasta`` to your own.
+2. Change the output dataset name from ``idaho`` to your own. Note the name restrictions.
+3. Change the output dataset title to any description.
+4. Change the subsampling scheme name from ``idaho_scheme`` to your own. Note the name restrictions.
+5. Change the regional dataset from North America to an appropriate region for your custom focal data:
+
+   1. Go to https://nextstrain.org/ncov/open/north-america.
+   2. Using the Dataset selector, select ``north-america`` to see the other possible options (e.g. ``south-america``, ``africa``).
+   3. Replace ``north-america`` in each of the ``metadata`` and ``aligned`` URLs with your choice of regional dataset.
+
+Run the workflow
+----------------
+
+From within the ``ncov/`` directory, run the ``ncov`` workflow:
+
+.. code:: text
+
+   nextstrain build . --cores 4 --configfile ncov-tutorial/genomic-surveillance.yaml
```

Comment:
This command had a subtle issue where the combined metadata (`results/combined_metadata.tsv.xz`) and sequences (`results/combined_sequences_for_subsampling.fasta.xz`) already existed from the "custom data" run, so the workflow didn't re-run the steps to combine these files and include the newly defined inputs. As a result the workflow crashed when trying to run an `augur filter` query that referred to a nonexistent `background_data` column.

To fix the issue, I had to tell the workflow to rebuild everything with the `--forceall` flag:

```
nextstrain build . --forceall  --configfile ncov-tutorial/genomic-surveillance.yaml
```

This seems like a Snakemake bug or a bug in our workflow; the new "background_data" entry and updated contents of the "custom_data" files should trigger a rebuild of files that depend on them.

Reply:
Still looking into the DAG issue with Snakemake, but noting that this build ran in 42 min on 3 CPUs with the config in ncov-tutorial@cb2f69fe413a6b979687b52b897f6fdd2a8c4da9.

Reply:
Ran [updated config](https://github.com/nextstrain/ncov-tutorial/blob/57a8d91021396ab72d077a5041b28a06d6abfa0e/genomic-surveillance.yaml) in 27 min on an M1 Mac using 8 cores (native runtime).

Reply:
@huddlej I am running this using Docker runtime on my computer (very slowly due to nextstrain/docker-base#35), but I am at the `refine` step and can see that this already ran successfully:

> python3 scripts/combine_metadata.py --metadata results/sanitized_metadata_reference_data.tsv.xz results/sanitized_metadata_custom_data.tsv.xz results/sanitized_metadata_background_data.tsv.xz --origins reference_data custom_data background_data --output results/combined_metadata.tsv.xz 2>&1 | tee logs/combine_input_metadata.txt

So, I don't think I am able to reproduce this 😕